### DISCUSSION TITLE
make revolt look and be more like discord

### DISCUSSION THREAD

**Discussion by pinkjimrussiagirl** [stoatchat/discussions#613] [Feature Suggestions]
please add all the current discord features to revolt (except the subscriptions), such as role icons, gradient roles, profile colors, profile wallpapers, user profile cards, redesigned layout, option to use the same layout as discord’s grid view layout, custom server profiles, redesigned roles and channels, extra options for channel types, video streaming in calls and voice channels, and an app for ios and ipados

**Comment by Rexogamer** [comment]
In order:
- should be a separate issue, think this might already be one 
- that isn't a discord feature? also we already have them, albeit currently only settable via the API
- see bullet 1
- wdym? we already support banners, unless you mean those profile picture decorations
- planned, work has started on this one in the new client 
- this is rather vague and I'd like to note that our intentions are not to build an open source carbon copy of Discord + we're working on a pretty big redesign 
- if you mean for the server list then maybe? but this kinda seems like copying them for the sake of it 
- see bullet 1/5
- this is also too vague to be particularly useful; what exactly are you looking for beyond "copy Discord"
-  this is a much broader feature; we have plans for a couple but I wouldn't necessarily say we intend to add every one Discord has, at least at this stage
- planned but blocked by the voice server rewrite
- a native iOS app is currently in early development 

I'll also again emphasise that some of these just feel like copying Discord for the sake of copying Discord - while we have taken inspiration from them in several ways, our intent is not to just make "Discord but open source" but to instead build a modern chat platform that will naturally differ in several ways. 

Finally, I'll also mention that, if you have some CSS knowledge, you can create a custom theme to make the app look more like Discord (in fact, there is at least 1 Discord-styled theme available via the Theme Store) and that we allow custom clients, which can be built to resemble all sorts of things.

**Reply by pinkjimrussiagirl** [reply]
the profile colors i meant is like in this screenshot
![IMG_0305_Original](https://github.com/revoltchat/revolt/assets/111091825/8ce8d0ea-5142-4971-8782-18592e8779cc)

also the second bullet where you said about if role gradients were already added, they aren’t added because i only see an option for a single color instead of multiple colors

also is there a way i can change the shapes of the status icons so i can put my own shapes on them instead of them being circles with solid color

also try adding custom server profiles because there is a permission setting in the role permissions but i don’t see the edit user profile option when i right click on someone

**Reply by Rexogamer** [reply]
- I wasn't confused about the profile colours, I was confused about the "profile wallpapers"
- To repeat: "albeit currently only settable via the API" - you can't create them in the app right now, but you can *see* them
- This is also planned as a built in feature
- Again, some work on user cards - and, by extension, server profiles - will happen in the new client

**Reply by pinkjimrussiagirl** [reply]
oh, but basically the profile wallpapers would be the same place as the profile colors except where you can upload an image, also the profile colors would do gradients as well

also is there an image showing what the new client looks like

**Reply by Rexogamer** [reply]
The new client is currently still in development and we do not have any public screenshots at this time.

**Reply by pinkjimrussiagirl** [reply]
ok, also is there a way to change the shapes of the status icons using css so i can make them look more like the discord ones in my custom css theme

**Reply by Rexogamer** [reply]
I *think* there might be a way? I'm not sure, though

**Reply by pinkjimrussiagirl** [reply]
on guilded it letted me change the status icons to any that i liked the look of but when i tried doing it on here it just stayed as the default dots no matter what i set the css to

**Reply by pinkjimrussiagirl** [reply]
also another reason why i’m trying to change the shapes of the status icons is because i prefer the look of the discord status icons better than the default ones

here is what i currently have in my xml when i tried changing the shapes:
```
/* Online */
:is(circle[Attributes Style])[style="fill: var(--status-online)"] {
    background-image: url("data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 1 1'><circle fill='white' cx='0.5' cy='0.5' r='0.5'/></svg>");
    background-color: hsl(139, 47.3%, 43.9%) !important;
}
/* Do not disturb */
:is(circle[Attributes Style])[style attribute="fill: var(--status-busy)"] {
    background-image: url("data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 1 1'><mask id='m'><circle fill='white' cx='0.5' cy='0.5' r='0.5'/><rect fill='black' x='0.125' y='0.375' width='0.75' height='0.25' rx='0.125' ry='0.125'/></mask><rect fill='black' x='0' y='0' width='1' height='1' mask='url(%23m)'/></svg>");
    background-color: hsl(359, 82.6%, 59.4%) !important;
}
/* Away */
:is(circle[Attributes Style])[style attribute="fill: var(--status-away)"] {
    background-image: url("data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 1 1'><mask id='m'><circle fill='white' cx='0.5' cy='0.5' r='0.5'/><circle fill='black' cx='0.25' cy='0.25' r='0.375'/></mask><rect fill='black' x='0' y='0' width='1' height='1' mask='url(%23m)'/></svg>");
    background-color: hsl(38, 95.7%, 54.1%) !important;
}
/* Streaming */
:is(circle[Attributes Style])[style attribute="fill: var(--status-streaming)"] {
    background-image: url("data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 1 1'><mask id='m'><circle fill='white' cx='0.5' cy='0.5' r='0.5'/><polygon fill='black' points='0.35,0.25 0.78301275,0.5 0.35,0.75'/></mask><rect fill='black' x='0' y='0' width='1' height='1' mask='url(%23m)'/></svg>");
    background-color: hsl(262, 46.8%, 39.8%) !important;
}
/* Offline */
:is(circle[Attributes Style])[style attribute="fill: var(--status-invisible)"] {
    background-image: url("data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 1 1'><mask id='m'><circle fill='white' cx='0.5' cy='0.5' r='0.5'/><circle fill='black' cx='0.5' cy='0.5' r='0.25'/></mask><rect fill='black' x='0' y='0' width='1' height='1' mask='url(%23m)'/></svg>");
    background-color: hsl(214, 9.9%, 50.4%) !important;
}
```

i think i might not be doing it right

**Comment by ghost** [comment]
so the post isnt bait?

**Comment by pinkjimrussiagirl** [comment]
> I _think_ there might be a way? I'm not sure, though

another reason why i’m trying to change the shapes of the status icons is because i prefer the look of the discord status icons better than the default ones

here is what i currently have in my xml when i tried changing the shapes:
```
/* Online */
:is(circle[Attributes Style])[style="fill: var(--status-online)"] {
    background-image: url("data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 1 1'><circle fill='white' cx='0.5' cy='0.5' r='0.5'/></svg>");
    background-color: hsl(139, 47.3%, 43.9%) !important;
}
/* Do not disturb */
:is(circle[Attributes Style])[style attribute="fill: var(--status-busy)"] {
    background-image: url("data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 1 1'><mask id='m'><circle fill='white' cx='0.5' cy='0.5' r='0.5'/><rect fill='black' x='0.125' y='0.375' width='0.75' height='0.25' rx='0.125' ry='0.125'/></mask><rect fill='black' x='0' y='0' width='1' height='1' mask='url(%23m)'/></svg>");
    background-color: hsl(359, 82.6%, 59.4%) !important;
}
/* Away */
:is(circle[Attributes Style])[style attribute="fill: var(--status-away)"] {
    background-image: url("data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 1 1'><mask id='m'><circle fill='white' cx='0.5' cy='0.5' r='0.5'/><circle fill='black' cx='0.25' cy='0.25' r='0.375'/></mask><rect fill='black' x='0' y='0' width='1' height='1' mask='url(%23m)'/></svg>");
    background-color: hsl(38, 95.7%, 54.1%) !important;
}
/* Streaming */
:is(circle[Attributes Style])[style attribute="fill: var(--status-streaming)"] {
    background-image: url("data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 1 1'><mask id='m'><circle fill='white' cx='0.5' cy='0.5' r='0.5'/><polygon fill='black' points='0.35,0.25 0.78301275,0.5 0.35,0.75'/></mask><rect fill='black' x='0' y='0' width='1' height='1' mask='url(%23m)'/></svg>");
    background-color: hsl(262, 46.8%, 39.8%) !important;
}
/* Offline */
:is(circle[Attributes Style])[style attribute="fill: var(--status-invisible)"] {
    background-image: url("data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 1 1'><mask id='m'><circle fill='white' cx='0.5' cy='0.5' r='0.5'/><circle fill='black' cx='0.5' cy='0.5' r='0.25'/></mask><rect fill='black' x='0' y='0' width='1' height='1' mask='url(%23m)'/></svg>");
    background-color: hsl(214, 9.9%, 50.4%) !important;
}
```

i think i might not be doing it right
